### DISCUSSION TITLE
Removing foreign key constraints from create_permissions_table

### DISCUSSION THREAD

**Discussion by mlobalobal** [spatie/laravel-permission#2258] [Q&A] [answered]
Hi all,

Im trying to integrate my laravel application with planet scale. to use the planetscale, we cant have the foreign key constraints,

For my own migration file its quite simple to remove the constraints, but currently i don't know how to remove it for spatie-permission,

On installation, spatie-permission will generate a create_permission_tables table, and i need to remove the constraints on this code

```
$table->foreign(PermissionRegistrar::$pivotRole)
                ->references('id') // role id
                ->on($tableNames['roles'])
                ->onDelete('cascade');
```

I simply replace this code with this line `$table->foreignId("role_id")`, and it doesnt work, maybe because it pivot table, since I'm new with laravel, i have no idea what to do,

Can some one show me the right way to do this.

Thanks a lot

**Comment by angeljqv** [comment]
> I'm new with laravel

[Migrations](https://laravel.com/docs/9.x/migrations)

**Comment by drbyte** [comment, marked as answer]
NOTE: WITHOUT FOREIGN KEY CONSTRAINTS ENFORCED, YOUR DATA WILL LOSE INTEGRITY. The package enforces "some" but NOT ALL relationships internally. It DOES rely on the foreign key constraints to maintain consistency and to delete related records.

That said ...

To prevent the migration from setting foreign key relationships you can delete the `$table->foreign()....;` command from the migration file before running it. 
or
If you need to remove the foreign key constraint after the migration is already run, create a new migration to remove it.
The command in the migration would be something like: `$table->dropForeign('name_of_foreign_key_here');`. Repeat for each relationship.

**Comment by abdhashem** [comment]
You could simply use:

```
Schema::table('table_name', function (Blueprint $table) {
    $table->dropForeign(['column_name']);
});
```
